### PR TITLE
Handling estimation tasks with 0 shot

### PR DESCRIPTION
**Pull request by jgonthier**
Now, estimation tasks with 0 shots are treated like constant tasks and evaluated without actually submitting to a backend.

### CODE REVIEW

File: src/python/zquantum/core/estimation/_estimation.py [hunk 1]
```diff
@@ -215,62 +214,67 @@ def split_constant_estimation_tasks(
...
     for task in estimation_tasks:
         if len(task.operator.terms) > 1 or () not in task.operator.terms.keys():
-            raise RuntimeError(
-                "evaluate_constant_estimation_tasks received an EstimationTask "
-                "that contained a non-constant term."
-            )
-        coefficient = task.operator.terms[()]
+            if task.number_of_shots > 0:
+                raise RuntimeError(
+                    "An EstimationTask required shots but was classified as a non-measured task"
```
Comment (max-radin) [inline src/python/zquantum/core/estimation/_estimation.py:271]: This line is too long to pass the style checks. Each line should be 88 characters or less.

File: src/python/zquantum/core/estimation/_estimation.py [hunk 2]
```diff
@@ -298,13 +302,13 @@ def estimate_expectation_values_by_averaging(
...
+        estimation_tasks_not_measured,
         indices_to_measure,
-        indices_for_constants,
-    ) = split_constant_estimation_tasks(estimation_tasks)
+        indices_not_to_measure,
+    ) = split_estimation_tasks_to_measure(estimation_tasks)
 
-    expectation_values_for_constants = evaluate_constant_estimation_tasks(
-        estimation_tasks_for_constants
+    expectation_values_for_constants = evaluate_non_measured_estimation_tasks(
```
Comment (max-radin) [inline src/python/zquantum/core/estimation/_estimation.py:310]: I'd suggest renaming `expectation_values_for_constants` to `estimation_tasks_not_measured` for clarity

**Review by max-radin — CHANGES_REQUESTED** [review]
Looks good! Just left a couple minor style comments.

File: src/python/zquantum/core/estimation/_estimation.py (hunk 1 above)
Comment (jgonthier) [reply src/python/zquantum/core/estimation/_estimation.py]: Fixed in 7271c6240983bb6beaef7c480e7deb773e8ad567

File: src/python/zquantum/core/estimation/_estimation.py (hunk 2 above)
Comment (jgonthier) [reply src/python/zquantum/core/estimation/_estimation.py]: Fixed in 7271c6240983bb6beaef7c480e7deb773e8ad567

**Review by max-radin — APPROVED** [review]

**Pull request merged**